pkg/controller/worker/machineset_vmo.go

```diff
@@ -172,6 +169,75 @@ func cleanupOrphanVMODependencies(ctx context.Context, client azureclient.Vmss,
 	return nil
 }
 
+func (w *workerDelegate) determineWorkerPoolVmoDependency(ctx context.Context, infrastructureStatus *azureapi.InfrastructureStatus, workerStatus *azureapi.WorkerStatus, workerPoolName string) (*azureapi.VmoDependency, error) {
+	if !azureapihelper.IsVmoRequired(infrastructureStatus) {
+		return nil, nil
+	}
+
+	// First: Lookup the vmo dependency for the worker pool in the worker status.
+	var dependencyInStatus *azureapi.VmoDependency
+	for _, dep := range workerStatus.VmoDependencies {
+		if dep.PoolName != workerPoolName {
+			continue
+		}
+		if dependencyInStatus != nil {
+			return nil, fmt.Errorf("found more then one vmo dependencies for workerpool %s in the worker provider status", workerPoolName)
+		}
+		runVar := dep
```

Comment:
If I understand correctly this is done because `dep` is reassigned in every loop iteration and you need to snapshot the value. In this case the purpose of `runVar` is not very clear  because the name is not clear and you would need to be aware of the loop pitfall to understand why it is there. 

Is maybe something like this more clear? 
```go
	for _, d := range workerStatus.VmoDependencies {
                dep := d
		if dep.PoolName != workerPoolName {
			continue
		}
		if dependencyInStatus != nil {
			return nil, fmt.Errorf("found more then one vmo dependencies for workerpool %s in the worker provider status", workerPoolName)
		}
		dependencyInStatus = &dep
	}
```

Reply:
Exactly, I need to capture the value. I need  define an explicit variable here otherwise `dependencyInStatus` would always point to the current `dep` value. I agree the name `runVar` is not optimal, but actually I don't need to do it for loop cycle in the beginning.

Reply:
> but actually don't need to do it for loop cycle in the beginning

Yes, the code is pretty much optimal as it was. My suggestion was only to prevent this common pitfall in a future contribution. It is just one of those "easy-to-miss" things.

Reply:
How about renaming the `runVar` in `depSnapshot` or something like this to make it easier to understand why we need this step?

Reply:
I approved anyway because this is clearly a matter of preference, but I like this suggestion a lot.

Reply:
Let's have this small improvement before we go on :)

Reply:
Done.